### PR TITLE
Blackbox test gen

### PR DESCRIPTION
**Pull request by rgrinberg**
As discussed before, this is an attempt at a saner way to specify our test matrix. I'm not fully happy with the result as there's a lot of funny boilerplate to generate s-expressions. On the plus side, we don't have to worry about the formatting of the sexps.

Some more notes:

* The platform module isn't pretty, but we need to share a bit of code between this and cram.mll. Better ideas here are welcome. In the future, we could unify -skip-versions and -skip-platforms into something sexp based and transparent.

* I haven't yet organized all the tests correctly. That will have to be done once the windows testing PR is revived.

* A less invasive solution is to keep the old tests, but manually add `-skip-platform` everywhere and perhaps add a `-with-external-deps` flag. I think this is also acceptable, but then we don't get to have nice "grouped" aliases for development.

### CODE REVIEW

**Comment by rgrinberg** [comment]
@diml I'm not really in love with this PR but there's some urgency in getting the configurator tests to run on windows.

I can also just commit jbuild.inc as the jbuild file directly and remove the generating script. At least this way we have the correct arguments for every runner.

File: test/blackbox-tests/test-cases/github25/run.t [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+The build itself uses only "plop.ca-marche", which doesn't have this
+problem. So jbuilder shouldn't crash because of "plop.ca-marche-pas"
+
+We need ocamlfind to run this test
+
+  $ jbuilder build @install --display short --only hello
+  File "root/jbuild", line 5, characters 14-28:
+  Error: Library "plop.ca-marche" not found.
+  Hint: try: jbuilder external-lib-deps --missing --only-packages hello @install
```
Comment (diml) [inline test/blackbox-tests/test-cases/github25/run.t]: Is this change expected?

**Comment by diml** [comment]
Ok, we can merge it now. I'm not sure about some of the changes to the tests though

File: test/blackbox-tests/test-cases/github25/run.t (hunk 1 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/github25/run.t]: Yeah, it's to get rid of the `chdir` in the test runner. What was the purpose of the original `chdir`, seems like the test works the same way if we don't change the root.

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/github25/run.t]: @diml is the above ok?

Comment (diml) [reply test/blackbox-tests/test-cases/github25/run.t]: Sorry, my review comment got lost...

Basically the error shouldn't be there. In the old test there was a `(setenv OCAMLPATH ../findlib ...)`, did you keep something similar? Alternatively, we can just set the variable here directly for simplicity

Comment (diml) [reply test/blackbox-tests/test-cases/github25/run.t]: I can't see the commit :(

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/github25/run.t]: Oops. Check again.

**Comment by diml** [comment]
Ok, LGTM

**Comment by diml** [comment]
Merging this now as I want to use the one alias per test to debug an issue with ocaml on osx

**Pull request merged**